File: 205. Isomorphic Strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        if(s == null || t == null)
            return false;
        if(s.length() != t.length())
            return false;
        Map<Character, Character> hash = new HashMap<Character, Character>();
        for(int i = 0; i < s.length(); i++){
            if(hash.containsKey(s.charAt(i))){
                if(hash.get(s.charAt(i)) != t.charAt(i))
                    return false;
            }
            else{
                if(hash.containsValue(t.charAt(i)))
                    return false;
                hash.put(s.charAt(i), t.charAt(i));
            }
        }
        return true;
    }
}

File: 205. Isomorphic Strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char, int> hash1, hash2;
        for(int i = 0; i < s.size(); i++){
            if(hash1[s[i]] != hash2[t[i]])
                return false;
            hash1[s[i]] = i + 1;
            hahs2[t[i]] = i + 1;
        }
        return true;
    }
};